___tests___/pet.test.js

```diff
@@ -0,0 +1,158 @@
+const Pet = require('../src/pet')
+
+describe('constructor', () => {
+    const pet = new Pet('Fido');
+    
+    it('returns an object', () => {
+        expect(new Pet('Fido')).toBeInstanceOf(Object);
+    });
+    it('sets the name property', () => {
+        expect(pet.name).toEqual('Fido');
+    });
+    it('has an initial age of 0', () => {
+        expect(pet.age).toEqual(0);
+    });
+});
+
+describe('growUp', () => {
+    it('increments age by 1', () => {
+        const pet = new Pet('Fido');
+
+        expect(pet.age).toEqual(0);
+        pet.growUp();
+        expect(pet.age).toEqual(1); 
+    });
+    it('increase hunger by 5', () => {
+        const pet = new Pet('Fido');
+
+        expect(pet.hunger).toEqual(0);
+        pet.growUp();
+        expect(pet.hunger).toEqual(5); 
+    });
+    it('decrease fitness by 3', () => {
+        const pet = new Pet('Fido');
+
+        expect(pet.fitness).toEqual(10);
+        pet.growUp();
+        expect(pet.fitness).toEqual(7);
+    });
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+        pet.age = 30;
+
+        expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    });
+});
+
+describe('walk', () => {
+    const pet = new Pet('Fido');
+
+    it('increase fitness by 4', () => {
+        pet.fitness = 4;
+        pet.walk();
+
+        expect(pet.fitness).toEqual(8);
+    });
+    it('fitness level should never go above 10', () => {
+        pet.fitness = 8;
+        pet.walk();
+
+        expect(pet.fitness).toEqual(10);
+    });
+    it('throws an error if the pet is not alive', () => {
+        pet.fitness = 0;
+
+        expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    });
+});
+
+describe('feed', () => {
+    const pet = new Pet('Fido');
+    it('decreases hunger by 3', () => {
+        pet.hunger = 3;
+        pet.feed();
+
+        expect(pet.hunger).toBe(0);
+    })
+    it('hunger should never go below 0', () => {
+        pet.hunger = 2;
+        pet.feed();
+
+        expect(pet.hunger).toBe(0)
+    });
+    it('throws an error if the pet is not alive', () => {
+        pet.hunger = 10;
+
+        expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    });
+});
+
+describe('checkUp', () => {
+    const pet = new Pet('Fido');
```

Comment:
This describe block is a good candidate for showing why its important to use beforeEach to keep each test separate. If you remove `const pet = new Pet('Fido');` on line 99 and then re run the tests, they will fail. This is because an already altered instance of `Pet` is being tested in the previous test, which can lead to wrongly passing/failing tests.

If you use the beforeEach as described above this will prevent this from happening and keep all your tests separate